### PR TITLE
Fix ATS score UI/logging drift and expose possible max score

### PR DESCRIPTION
**Pull request by damilola-elegbede**
## Summary
- fix ATS score card rendering/safety issues (invalid values, target-arrow disconnect)
- align resume generation logging with edit-aware scoring and breakdown math
- enforce immutable fields in modify-change responses
- preserve resolved JD text for URL-based analyses when persisting generation logs
- add shared score utilities and use one formula path for `currentScore` and `possibleMaxScore`
- expose `possibleMax`/`possibleMaxScore` in generation summary payloads used by API/MCP
- add regression tests for score sanitization and boundary-aware edited keyword matching

## Validation
- npm run typecheck
- npm test -- tests/lib/score-utils.test.ts tests/lib/edit-aware-rescoring.test.ts tests/components/admin/compatibility-score-card.test.tsx tests/app/api/resume-generator-log.test.ts tests/app/api/v1-resume-generations.test.ts tests/mcp/tools/resume-generations.test.ts tests/app/api/resume-generator.test.ts

## Summary by CodeRabbit

* **New Features**
  * Added fields to expose current and maximum possible compatibility scores; URL-resolved job descriptions improve analysis metadata and score displays.
  * Improved keyword-matching for more accurate impact calculations.

* **Bug Fixes**
  * Sanitized and capped score values and breakdowns to prevent invalid or out-of-range displays.
  * Guard added to prevent AI-generated edits from altering immutable fields.

* **Tests**
  * Expanded tests for score utilities, normalization, and non-finite value handling.

### CODE REVIEW

File: src/lib/resume-scoring.ts [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+    const escaped = normalizedKeyword.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    return new RegExp(`\\b${escaped}\\b`, 'i').test(normalizedText);
+  }
+
+  // For multi-word phrases, allow flexible whitespace and punctuation boundaries.
+  const phrasePattern = normalizedKeyword
+    .split(/\s+/)
+    .map((part) => part.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))
+    .join('[\\s\\-/,]+');
+  return new RegExp(`\\b${phrasePattern}\\b`, 'i').test(normalizedText);
```
[the remark this answers is not in the document]
Comment (damilola-elegbede) [reply src/lib/resume-scoring.ts]: Implemented and pushed follow-up fixes in commit b4ddaf2:\n- switched regex boundaries to `(?<!\\w)` / `(?!\\w)` in `resume-scoring` for punctuation-heavy tokens\n- added `normalizeEstimatedCompatibility(...)` in `resume-generator/log` to sanitize partial/invalid inputs\n- aligned projected score semantics to base+delta\n- deduplicated possible-max fallback in admin/v1 generation routes\n- reused shared `sanitizeScoreValue` in `ats-scorer`\n- added regression tests for partial estimatedCompatibility normalization and punctuation-token matching\n\nValidation run: targeted Vitest suites + `npm run typecheck` all pass.\n\n@coderabbitai resolve
